### PR TITLE
Customizable log4j properties for kafka brokers.

### PR DESCRIPTION
**Pull request by cmars**
Drive-by fix for broker_id = 0 edge case in charm.
Update test bundle charm revisions to latest.

### CODE REVIEW

File: charm/kafka/lib/charms/layer/kafka.py [hunk 1]
```diff
@@ -127,6 +127,13 @@ def install(self, zk_units=[], log_dir='logs'):
             context=context
         )
 
+        log4j_file = os.path.join(KAFKA_SNAP_DATA, 'log4j.properties')
+        if config.get('log4j_properties'):
+            with open(log4j_file, 'w') as f:
+                print(config['log4j_properties'], file=f)
```
Comment (alnvdl) [inline charm/kafka/lib/charms/layer/kafka.py:133]: Out of curiosity: why `print(file=...)` insteaf of simply `f.write`?

File: charm/kafka/config.yaml [hunk 2]
```diff
@@ -16,6 +16,12 @@ options:
     description: |-
       Default replication factor for automatically created topics
       (default.replication.factor)
+  log4j_properties:
+    default: ""
+    type: string
+    description: |-
+      Contents of a custom log4j.properties file for the Kafka brokers. If
```
Comment (alnvdl) [inline charm/kafka/config.yaml:23]: What is the default? From the code below, it looks like if it's empty and the file exists, it will be removed. Otherwise, if it's empty and the file does not exist, nothing happens.

**Review by alnvdl — APPROVED** [review]
LGTM, with a couple of questions.

File: charm/kafka/lib/charms/layer/kafka.py (hunk 1 above)
Comment (cmars) [reply charm/kafka/lib/charms/layer/kafka.py:133]: f.write expects a `bytes` argument, but the config value is a utf8 `str`. Alternatively I could have explicitly encoded the `str` to a `bytes` as utf-8.

File: charm/kafka/config.yaml (hunk 2 above)
Comment (cmars) [reply charm/kafka/config.yaml:23]: The changes to the snap wrapper will cause the custom log4j.properties file to be used if it exists -- otherwise the kafka startup script will use a default configuration, `/opt/kafka/config/log4j.properties`.

So we want to remove that custom file if a custom log4j_properties was set in the past and a custom log4j.properties file was written, but it is no longer set (config change from non-empty string to empty string, `juju config --reset kafka log4j_properties`, etc)
